dbt_invoke/internal/_utils.py

```diff
@@ -161,16 +163,36 @@ def dbt_ls(
                 default_arguments.append(f'{get_cli_kwargs(resource_type=rt)}')
     default_arguments = ' '.join(default_arguments)
     arguments = get_cli_kwargs(**kwargs)
-    all_arguments = f'{default_arguments} {arguments} --output {output}'
-    command = f"dbt ls {all_arguments}"
+    dbt_command_cli_args = f'{default_arguments} {arguments} --output {output}'
+    dbt_global_cli_args = get_cli_kwargs(**DBT_GLOBAL_ARGS)
+    command = f"dbt {dbt_global_cli_args} ls {dbt_command_cli_args}"
     logger.debug(f'Running command: {command}')
     result = ctx.run(command, hide=hide)
     result_lines = result.stdout.splitlines()
-    if output == 'json':
-        result_lines = [
-            json.loads(result_json) for result_json in result_lines
-        ]
-    return result_lines
+    result_lines_filtered = list()
+    for line in result_lines:
+        # Because we set the dbt global arg "--log-format json", if
+        # line is valid json then it may be an actual result or it
+        # may be some other output from dbt, like a warning.
+        try:
+            line_dict = json.loads(line)
+        # If line is not valid json, then it should be an actual
+        # result. This is because even when the "dbt ls" command
+        # arg "--output" is not set to json, non-result logs will
+        # still be in json format (due to the dbt global arg
+        # "--log-format json").
```

Comment:
Want to understand this comment and the try/except-- does this mean that 
1. if the command is run with `--output path`, result logs will _not_ be in json format and can just be appended result_lines_filtered, but non-result logs will still be in json format? 
2. if the command is run with `--output json`, both result logs and non-result logs will be in json format?

Reply:
Exactly